TEMP_CCAAS_TUTORIAL.md

```diff
@@ -0,0 +1,66 @@
+# Running Chaincode as Service with the Test Network
+
+The chaincode-as-a-service feature is a very useful and practical way to run 'Smart Contracts'. Traditionally the Fabric Peer has taken on the role of orchestrating the complete lifecycle of the chaincode. It required access to the Docker Daemon to create images, and start containers. Java, NodeJS and Go where explicitly known to the peer including how they should be built and started. 
+
+As a result this makes it very hard to deploy into K8S style environments, or to run in any form of debug mode. Plus the code is being rebuilt by the peer so production deployments can have some degree of uncertainty about what dependencies have been pulled in.
+
+Chaincode-as-service requires you to orchestrate the build and deployment phase yourself. Whilst this is an additional step, it gives control back. The Peer stills requires a 'chaincode package' to be installed. In this case this doesn't contain code, but the information about where the chaincode is hosted. (Hostname,Port,TLS config etc)
+
+## Prereqs
+
+We need to use the latest 2.4.1 release as this contains some improvements to make this process easier. The core functionality is available in earlier releases, but isn't quite as easy to use. 
+
+- ... clone samples repo etc.. TBC
+
+
+## Start the test network
+
+It's useful to have two terminal windows open, one for starting the Fabric Network, and a second for monitoring all the docker containers.
+
+In your 'monitoring' window, run this
+
+```bash
+# from the fabric-samples repo
+./test-network/monitordocker.sh
+```
+
+All the containers on the `fabric-test` network will be monitored; remember to terminate this as it starts a docker image on the `fabric-test` network.
+
+
+In the 'Fabric Network' window, start the test network
+
+```bash
+cd test-network
+./network.sh up createChannel
+```
+
+Once this has started, you can run the 'Chaincode-as-a-service'. This will do the following.
+
+- Build a docker container of the contract, in this example `/asset-transfer-basic/chaincode-typescript`
+- Install, Approve and Commit a chaincode definition. This is unchanged, but the chaincode package contains connection inforamtion, not code
+- Start the docker container containing the contract
+
+```
+./network.sh deployCCAAS  -ccn basic -ccp ../asset-transfer-basic/chaincode-typescript -cport 9999
+```
+
+This is very similar to the `deployCC` command, it needs the name, and path. But also needs to have the port the chaincode container is going use. As each container is on the `fabric-test` network, you might wish to alter this so there are no collisions with other chaincode containers.
```

Comment:
How many instances of ccaas container run? One for each peer? They all run on port 9999?